src/ring_theory/ideal/quotient.lean

```diff
@@ -410,4 +411,47 @@ noncomputable def quotient_inf_ring_equiv_pi_quotient [finite ι] (f : ι → id
 
 end chinese_remainder
 
+/-- The product over `fin 2` of some rings is just the cartesian product of these rings. -/
+@[simps]
+def _root_.ring_equiv.fin_two (R : fin 2 → Type*) [Π i, semiring (R i)] :
+  (Π (i : fin 2), R i) ≃+* R 0 × R 1 :=
+{ to_fun := pi_fin_two_equiv R,
+  map_add' := λ a b, rfl,
+  map_mul' := λ a b, rfl,
+  .. pi_fin_two_equiv R }
```

Comment:
I don't think this belongs in `ring_theory/ideal/quotient`? Maybe `algebra/ring/pi` or `algebra/ring/prod`?
Also, I don't really understand this name. Maybe `ring.pi_fin_two_equiv`?

Reply:
I couldn't find a nice place for this lemma (the join of the required imports would be `algebra.module.linear_map`...). Shall I just make a new file, named something like `algebra.ring.fin` to match `logic.equiv.fin`?

I'd like to keep it in the `ring_equiv` namespace since this equiv exists for many structures, so how about `ring_equiv.pi_fin_two`?

Reply:
Another argument for the name `ring_equiv.pi_fin_two`: [`linear_equiv.pi_fin_two`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/pi.html#linear_equiv.pi_fin_two)

Reply:
Sure! I just thought that pi should be in the name.